spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -260,12 +269,23 @@ public void bindOutputs(ChannelBindingService channelBindingService) {
 			ChannelHolder channelHolder = channelHolderEntry.getValue();
 			String outputChannelName = channelHolderEntry.getKey();
 			if (channelHolderEntry.getValue().isBindable()) {
-				if (log.isDebugEnabled()) {
+			if (log.isDebugEnabled()) {
 					log.debug(String.format("Binding %s:%s:%s", this.channelNamespace, this.type, outputChannelName));
 				}
 				channelBindingService.bindProducer(channelHolder.getMessageChannel(), outputChannelName);
 			}
 		}
+		//Make sure to bind error channel as a message producer only if there is at least a single output bindable exists.
+		//i.e. binding error channel of the  outbound endpoints.
+		if (this.outputBindableExists) {
+			bindErrorChannel(channelBindingService);
+		}
+	}
+
+	private void bindErrorChannel(ChannelBindingService channelBindingService) {
```

Comment:
Maybe we can move this into a standalone Bindable implementation, `ConditionalOnProperty("spring.cloud.stream.bindings.error.destination"))`